### PR TITLE
Add test coverage for SearchActionFilter

### LINKED ISSUE #31: Improve test coverage
This project currently (November 22, 2022) has around 21% line coverage from tests. 

We should increase that significantly to get more confidence on future changes (and maybe uncover some lurking bugs along the way).

I'm going to suggest a target of 80% before we can call this issue "done".

### PR DESCRIPTION
**Pull request by msfroh**
Signed-off-by: Michael Froh <froh@amazon.com>

### Description
Adds unit tests for SearchActionFilter class (plus OpenSearchClient as collateral impact).

### Issues Resolved
#31 won't actually be "resolved", but this contributes.
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java [hunk 1]
```diff
@@ -107,7 +108,8 @@ public <Request extends ActionRequest, Response extends ActionResponse> void app
       // Source is returned in response hits by default. If disabled by the user, overwrite and enable
       // in order to access document contents for reranking, then suppress at response time.
       boolean suppressSourceOnResponse = false;
-      if (searchRequest.source().fetchSource() != null && !searchRequest.source().fetchSource().fetchSource()) {
+      if (searchRequest.source() != null && searchRequest.source().fetchSource() != null &&
+              !searchRequest.source().fetchSource().fetchSource()) {
```
Comment (macohen) [inline src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java:112]: Is it a problem that we have to reach so deep into the SearchRequest?  Not saying that we can do anything different here...

File: src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java [hunk 2]
```diff
@@ -143,14 +145,13 @@ private List<ResultTransformerConfiguration> getResultTransformerConfigurations(
...
             .stream()
-            .map(s -> s.getKey())
-            .collect(Collectors.toList()))
-        .flatMap(Collection::stream)
-        .collect(Collectors.toList())
-        .toArray(new String[0]);
+            .map(t -> t.getTransformerSettings()
+                    .stream()
+                    .map(Setting::getKey)
+                    .collect(Collectors.toList()))
```
Comment (macohen) [inline src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java:152]: Why this change? Is it because collect was being called twice unnecessarily?

File: src/main/java/org/opensearch/search/relevance/configuration/SearchConfigurationExtBuilder.java [hunk 3]
```diff
@@ -124,8 +124,10 @@ public int hashCode() {
     return Objects.hash(this.getClass(), this.resultTransformerConfigurations);
   }
 
-  public void setResultTransformers(final List<ResultTransformerConfiguration> resultTransformerConfigurations) {
-    this.resultTransformerConfigurations = resultTransformerConfigurations; }
+  public SearchConfigurationExtBuilder setResultTransformers(final List<ResultTransformerConfiguration> resultTransformerConfigurations) {
```
Comment (macohen) [inline src/main/java/org/opensearch/search/relevance/configuration/SearchConfigurationExtBuilder.java:127]: is there a different standard now for creating a "set" method that returns something? Should this be named differently?

**Review by macohen — COMMENTED** [review]
Just had a few questions, but looks good overall.

File: src/main/java/org/opensearch/search/relevance/configuration/SearchConfigurationExtBuilder.java (hunk 3 above)
Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/configuration/SearchConfigurationExtBuilder.java:127]: I normally write builders with `withXXX` methods for chaining, but the longstanding convention with builders in the codebase uses `setXXX` that returns the builder.

I disagree with the convention, but I need to follow it. (Well, I guess I don't *need* to, but I think consistency is more important in this case.)

File: src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java (hunk 2 above)
Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java:152]: Yeah... IntelliJ flagged it as a warning, effectively saying "Why are you taking two steps to convert this stream to an array when you can just do it in one step?" With a quick Alt-Enter, it fixed it for me.

File: src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java (hunk 1 above)
Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java:112]: Yeah... this is just how the core SearchRequest class is structured:

* SearchRequest holds (mostly optional) high-level flow-control type stuff (routing param, scroll state, timeout), plus a reference to the SearchSourceBuilder (which never actually builds a SearchSource, but 🤷 ).
* SearchSourceBuilder holds most of what we think of as "the search request" -- the query, highlight options, sort options, from+size, etc. including the FetchSourceContext.
* The FetchSourceContext effectively represents return field options -- `fetchSource` is a boolean where `true` means "Yes, return all fields" and `false` means "No, return no fields". You can also specify includes/excludes (which may include wildcards) to fetch a subset of fields.

**Review by macohen — APPROVED** [review]
